### ISSUE TITLE
sso doesn't report error if permission request is dismissed

### ISSUE THREAD

**Issue by David-Development** [nextcloud/Android-SingleSignOn#39]
If the user clicks outside the permission request dialog, the sso library doesn't return a proper error message.

**Comment by tobiasKaminsky** [comment]
Is this still valid?

**Comment by stefan-niedermann** [comment]
It hasn't been implemented yet. To be honest i didn't miss it, but if @David-Development still needs this, it can be implemented either by
- Directly displaying an Alert from the SSO library which says "You need to grant the permissions to use the SSO." or
- throw an Exception similar to the `AccountImportCancelledException` (https://github.com/nextcloud/Android-SingleSignOn/issues/124 and https://github.com/nextcloud/Android-SingleSignOn/pull/129) (preferred)

**Comment by tobiasKaminsky** [comment]
As this prevents sso from finishing, it is a good idea to throw AccountImportCancelledException :+1:

**Comment by stefan-niedermann** [comment]
Fine with it, just want to note that this was a breaking change because 3rd party clients which do not yet catch this exception at this place will break in case a user dismisses the permission

→ Requires a new major release

**Comment by David-Development** [comment]
I fully agree with Stefan. New major release and the Exceptions sounds great to me as well! 👍 

@stefan-niedermann @tobiasKaminsky fyi: Probably this issue is also related to: https://github.com/nextcloud/Android-SingleSignOn/issues/202

**Comment by stefan-niedermann** [comment]
As far as i can see, this has to be fixed in the `Files` app.

In the SSO lib `AccountImporter#onActivityResult` the result code `RESULT_OK` is given, but it should be `RESULT_CANCELED`. Then the wanted `AccountImportCancelledException` would be thrown.

I also wouldn't consider this a breaking change anymore because the method already `throws AccountImportCancelledException` since quite a long time.

@tobiasKaminsky Should i transfer this issue in the `Files` app repository or open a separate one?
